HHS/Head-Start-TTADP
# [PROD] Added Close/Suspend Reason Modal, Merge Two Recipients, Fix for Creator Role

## Description of change

**TTAHUB-656:** If a user attempts to Close or Suspend a Goal we need them to provide a reason.

**TTAHUB-705:** Merges recipients. They shouldn't be double listed on the Recipients TTA Record page with this change. Nightly download from HSES script has been updated (Arkansas Department of Human Services).

**TTAHUB-711** Font colors on our graphs weren't passing accessibility scanning. This change darkens the color on the labels on our graph's x & y axis's.

**TTAHUB-707** New database tables linking objectives & topics and objectives & resources

**TTAHUB706:** We encountered an issue when deploying the new creator role field to production. If a report was in NEEDS_ACTION state the user has no way of setting the required creator role. This prevented them from resubmitting the report. We also found out that the same issue existed if the user had one role and a report was in the DRAFT state.

## How to test

**TTAHUB-656:** Go to any recipient and view the 'Goals & Objectives' tab. When attempting to Close or Suspend a Goal you should be prompted to fill out a model with a reason and optional context.

Note: Before we push this PR the code to change the actual Goal status and reason will be commented out.

**TTAHUB-705:** Check that the RTR page for region 6 doesn't have double recipients.

**TTAHUB-711:** The graphs on the dashboard &  RTR still render correctly. They have a new, darker font color

**TTAHUB-707** Backend only, no way to test

**TTAHUB706:** Using a user with one role. Create a report that is in draft with no creator role. Saving the report should add the role. Create needs action report with no role resubmitting the report should set the creator role.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-656
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-705
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-707
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-711
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-706

## Checklists

### Every PR


~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy


### After merge/deploy
